web/cypress/integration/route/create-route-with-service-discovery-upstream.spec.js

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-undef */
+
+context('Create Route with Upstream', () => {
+  const selector = {
+    name: '#name',
+    description: '#desc',
+    discovery_type: '#discovery_type',
+    service_name: '#service_name',
+    upstreamSelector: '[data-cy=upstream_selector]',
+    nameSelector: '[title=Name]',
+    selectItem: '.ant-select-item-option-content',
+    drawer: '.ant-drawer-content',
+    monacoScroll: '.monaco-scrollable-element',
+    input: ':input',
+    deleteAlert: '.ant-modal-body',
+    notification: '.ant-notification-notice-message',
+  };
+
+  const data = {
+    deleteRouteSuccess: 'Delete Route Successfully',
+    submitSuccess: 'Submit Successfully',
+    deleteUpstreamSuccess: 'Delete Upstream Successfully',
+    weight: 1,
+    description: 'desc_by_autotest',
+    upstreamName: 'test_upstream',
+    routeName: 'test_route',
+    serviceName: 'test.cluster.local',
+  };
+
+  beforeEach(() => {
+    cy.login();
+  });
+
+  it('should create route with service discovery upstream', function () {
+    cy.visit('/');
+    cy.get('[role=menu]')
+      .should('be.visible')
+      .within(() => {
+        cy.contains('Route').click();
+      });
+    cy.contains('Create').click();
+
+    cy.contains('Next').click().click();
+    cy.get(selector.name).type(data.routeName);
+    cy.contains('Next').click();
+
+    // should enable Upstream input boxes after selecting Custom mode
+    cy.get(selector.upstreamSelector).click();
+    cy.contains('.ant-select-item-option-content', 'Custom').click();
+
+    // set service discovery
+    cy.get('[title="Node"]').click();
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('Service Discovery').click();
+    });
+
+    cy.get(selector.discovery_type).click();
+    cy.get(selector.selectItem).within(() => {
+      cy.contains('DNS').click();
+    });
+    cy.get(selector.service_name).type(data.serviceName);
+
+    cy.contains('Next').click();
+    cy.contains('Next').click();
+    cy.contains('Submit').click();
+    cy.contains(data.submitSuccess).should('be.visible');
+    cy.contains('Goto List').click();
+    cy.url().should('contains', 'routes/list');
+  });
+
+  it('should edit this route with upstream', function () {
+    cy.visit('/');
+    cy.contains('Route').click();
+    cy.get(selector.nameSelector).type(data.routeName);
+
+    cy.contains('Search').click();
+    cy.contains(data.routeName).siblings().contains('Configure').click();
+
+    cy.get(selector.name).should('value', data.routeName);
+    cy.contains('Next').click({
+      force: true,
+    });
+
+    cy.wait(1000);
+
+    // set another service discovery
+    cy.get(selector.discovery_type).click({ force: true });
```

Comment:
should assert whether the `discovery_type` is DNS

Reply:
Here, we can't actually get the value in select through cypress because antd encapsulates it. Even the actual input box behind it cannot get its value.